### PR TITLE
Adding clean target to ci-build target

### PR DESCRIPTION
**Pull request by algojack**
While building with Brice, we noticed that `make ci-build` doesn't clean automatically, and we probably want that by default. So we decided to add it.

Are there any times we don't want clean included?

### CODE REVIEW

File: scripts/release/mule/Makefile.mule [hunk 1]
```diff
@@ -4,7 +4,7 @@ PKG_DIR = $(SRCPATH)/tmp/node_pkgs/$(OS_TYPE)/$(ARCH)
 
 .PHONY: ci-clean ci-setup ci-build
 
-ci-clean:
+ci-clean: clean
```
Comment (algobarb) [inline scripts/release/mule/Makefile.mule:7]: What does clean do? Is it like `cleanWs()` which cleans the Workspace?

Comment (algojack) [reply scripts/release/mule/Makefile.mule:7]: @bricerisingalgorand a little help? I think it deletes temp files, but i'm not 100% sure.

Comment (bricerisingalgorand) [reply scripts/release/mule/Makefile.mule:7]: https://github.com/algorand/go-algorand/blob/master/Makefile#L252

Comment (bricerisingalgorand) [reply scripts/release/mule/Makefile.mule:7]: That tries to delete all build/generated files in the current repo

**Review by algobarb — APPROVED** [review]

**Review by egieseke — APPROVED** [review]
Good catch.

**Pull request merged**